app/src/main/java/com/chouten/app/data/WebviewHandler.kt

```diff
@@ -2,358 +2,291 @@ package com.chouten.app.data
+    
+    @JavascriptInterface
+    fun sendHTTPRequest(data: String) {
+        val self = this
+
+        runBlocking {
+            try{ 
+                self.postMessage(data) 
+            }catch(error: Exception){
+                val message = error.message;
+
+                if(message != null){
+                    val errorJSON = mapOf<String, String>(
+                        "result" to (message),
+                        "action" to "error"
+                    );
+                    self.callback(JSONObject(errorJSON).toString());
+                }else{
+                    self.callback("{'action': 'error', 'message': 'Error'}");
                 }
-                counter += 1
-                responseCode = _response.code
-                response = _response.body.bytes()
-                if (_response.isSuccessful && responseCode == 200) break
             }
-            response
-        } catch (e: Exception) {
-            LogLayer.addLogEntry(
-                LogEntry(
-                    title = "Failed to load data from ${request?.url}",
-                    message = e.stackTraceToString(),
-                    isError = true
-                )
-            )
-
-
-            e.printStackTrace()
-            return false
         }
+    }
 
+    suspend fun postMessage(message: String) {
+        val req = Mapper.parse<RequestOption>(message);
+        var self = this;
 
-        LogLayer.addLogEntry(
-            LogEntry(
-                title = "Loading data from ${request?.url}",
-                message = "Request Type: ${request?.method}\nUses API? ${codeblock.usesApi}",
-                isError = false
-            )
-        )
-
-
-        // The webview expects a Base64 encoded string
-        val encodedData = Base64.encodeToString(data, Base64.DEFAULT)
-        val decodedData = String(Base64.decode(encodedData, Base64.DEFAULT))
-
-        if (!codeblock.usesApi!!) {
-            webview.loadDataWithBaseURL(null, decodedData, "text/html; charset=utf-8", "br", null)
-        } else {
-            // We want to load a skeleton
-            // so that the webview is ready to go
-            // when we inject the data.
-
-            val sanitised =
-                decodedData
-                    // Replace the HTML ampersand values
-                    // e.g &#39; -> '
-                    .replace(Regex("&#\\d+;"), "")
-                    .replace("'", "")
-                    .replace('"', '\'')
-
-            // We want to buffer the sanitised data
-            // and print it to the logcat
-            // so that we can debug it.
-            val chunkSize = 1000
-            val chunks = sanitised.chunked(chunkSize)
-            chunks.forEachIndexed { index, chunk ->
-                Log.d("WebviewHandler", chunk)
-            }
+        if(req.action == "HTTPRequest" && req.url != null && req.headers != null){
+            
+            var responseText = "";
```

Comment:
This does not need to be mutable. `val responseText = if ....` is perfectly fine

Reply:
Converted it to val for now. Have to experiment with Kotlin one-liners more.